src/cxx_frontend/ast_exporter/NodeListSerializer.h

```diff
@@ -33,6 +33,12 @@ template <typename Stub, typename NodeSerializerImpl> class NodeListSerializer {
         m_orphanage.newOrphan<stubs::Loc>(), m_orphanage.newOrphan<Stub>());
     m_serializer.serialize(item, nodeOrphan.locOrphan.get(),
                            nodeOrphan.descOrphan.get());
+
+    //if(std::is_convertible<T, clang::Decl const *>::value
+    if constexpr(std::is_same<stubs::Decl, Stub>::value) if(nodeOrphan.descOrphan.get().isUnionNotInitialized())
+    {
+      m_orphans.pop_back();
+    }
```

Comment:
At this point the orphan was already constructed, which means that corresponding memory was already allocated for the orphan in the `capnp` message. The receiver of the message won't be able to decode the orphan (which is fine), but he allocated memory in the message would be wasted.

You could add a `Deleted` declaration to the declaration union in the `capnp` schema and set is as active member when a declaration is a deleted method. In `Verifast`, we can simply ignore these `Deleted` declarations by returning an empty list in `decl_translator`.

For defaulted methods it should be fine to have `UnionNotInitialized` as active union member because you already emit an error. `ast_translator` first checks if any errors were serialized before decoding the translation unit. If errors are found, decoding of the translation unit is skipped. Therefore, the `UnionNotInitialized` won't be translated.

Reply:
You meant like this?

Reply:
Looks good! Thanks!